app/v1/controllers/validators/AmendOtherReliefsRulesValidator.scala

```diff
@@ -0,0 +1,155 @@
+      annualPaymentsMade.traverse_(validateAnnualPaymentsMade),
+      zipAndValidate(qualifyingLoanInterestPayments, validateQualifyingLoanInterestPayments)
+    ).onSuccess(parsed)
+  }
+
+  private def validateNonDeductibleLoanInterest(nonDeductibleLoanInterest: NonDeductibleLoanInterest): Validated[Seq[MtdError], Unit] = {
+    import nonDeductibleLoanInterest._
+
+    combine(
+      customerReference.traverse_(validateCustomerRef(_, "/nonDeductibleLoanInterest/customerReference")),
+      resolveParsedNumber(reliefClaimed, None, Some("/nonDeductibleLoanInterest/reliefClaimed"))
+    )
+  }
+
+  private def validatePayrollGiving(payrollGiving: PayrollGiving): Validated[Seq[MtdError], Unit] = {
+    import payrollGiving._
+
+    combine(
+      customerReference.traverse_(validateCustomerRef(_, "/payrollGiving/customerReference")),
+      resolveParsedNumber(reliefClaimed, path = Some("/payrollGiving/reliefClaimed"))
+    )
+  }
+
+  private def validateQualifyingDistributionRedemptionOfSharesAndSecurities(
+      qualifyingDistributionRedemptionOfSharesAndSecurities: QualifyingDistributionRedemptionOfSharesAndSecurities)
+      : Validated[Seq[MtdError], Unit] = {
+    import qualifyingDistributionRedemptionOfSharesAndSecurities._
+
+    combine(
+      customerReference.traverse_(validateCustomerRef(_, "/qualifyingDistributionRedemptionOfSharesAndSecurities/customerReference")),
+      resolveParsedNumber(amount, path = Some("/qualifyingDistributionRedemptionOfSharesAndSecurities/amount"))
+    )
+  }
+
+  private def validatePostCessationTradeReliefAndCertainOtherLosses(
+      postCessationTradeReliefAndCertainOtherLosses: PostCessationTradeReliefAndCertainOtherLosses,
+      index: Int): Validated[Seq[MtdError], Unit] = {
+    import postCessationTradeReliefAndCertainOtherLosses._
+
+    val validatedCustomerReference =
+      customerReference
+        .traverse_(validateCustomerRef(_, s"/postCessationTradeReliefAndCertainOtherLosses/$index/customerReference"))
+
+    val validatedBusinessName =
+      businessName
+        .traverse_(validateFieldLength(_, s"/postCessationTradeReliefAndCertainOtherLosses/$index/businessName", BusinessNameFormatError))
+
+    val validatedDateBusinessCeased =
+      dateBusinessCeased
+        .traverse_(ResolveIsoDate(_, Some(DateFormatError), Some(s"/postCessationTradeReliefAndCertainOtherLosses/$index/dateBusinessCeased")))
+
+    val validatedNatureOfTrade =
+      natureOfTrade
+        .traverse_(validateCustomerRef(_, s"/postCessationTradeReliefAndCertainOtherLosses/$index/natureOfTrade", NatureOfTradeFormatError))
+
+    val validatedIncomeSource =
+      incomeSource
+        .traverse_(validateFieldLength(_, s"/postCessationTradeReliefAndCertainOtherLosses/$index/incomeSource", IncomeSourceFormatError))
+
+    val validatedAmount = resolveParsedNumber(amount, None, Some(s"/postCessationTradeReliefAndCertainOtherLosses/$index/amount"))
```

Comment:
there's a bit of a mix in this class of "validated" vals then being passed into combine(), and inlined validations being passed in. Ought to be one or the other really. I'd suggest the latter, just less code then.